app/src/main/kotlin/com/github/gotify/CoilInstance.kt

```diff
@@ -85,3 +90,24 @@ object CoilInstance {
         return sslSettings to loader
     }
 }
+
+private class BasicAuthInterceptor : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var request = chain.request()
+
+        // If there's no username, skip the authentication
+        if (request.url.username.isNotEmpty()) {
+            val basicAuthString = "${request.url.username}:${request.url.password}@"
+            val url = request.url.toString().replace(basicAuthString, "")
```

Comment:
Does this replace have a use, for me it works with and without? I've improved the placeholder handling and added logging for failing requests. Could you have a look at it?

Reply:
@cyb3rko should I merge this without your review?

Reply:
Sorry, there's a lot going on at the moment.
Hopefully I can find some time today to take a look.

Reply:
Your changes work on my side.  
I could have sworn I found edge cases where it wouldn't work without stripping the URL, but I can't reproduce them at the moment. The only thing that could be interesting is what happens if the response is still 401 even after applying basic auth. Does it always start the interceptor again and again and run into request loops?
With stripping the URL we can not have that problem at all because on the next try it does not have a `request.url.username` anymore.

Nevertheless, I'm fine with merging.

Reply:
Thanks for looking over it. It seems this behavior is required when using `okhttp3.Authenticator`

> ## Reactive Authentication
>
> Implementations authenticate by returning a follow-up request that includes an authorization
> header, or they may decline the challenge by returning null. In this case the unauthenticated
> response will be returned to the caller that triggered it.
>
> Implementations should check if the initial request already included an attempt to
> authenticate. If so it is likely that further attempts will not be useful and the authenticator
> should give up.

but the Interceptor doesn't do new requests, so I'll remove it from here.